segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperServiceRunner.java

```diff
@@ -253,10 +258,37 @@ private void initializeZookeeper() throws Exception {
     private void runBookies() throws Exception {
         log.info("Starting Bookie(s) ...");
         for (int bkPort : this.bookiePorts) {
-            this.servers.add(runBookie(bkPort));
+            this.servers.add(runBookieWithRetry(bkPort));
         }
     }
 
+    private BookieServerAndResources runBookieWithRetry(int bkPort) throws Exception {
+        int count = 0;
+        boolean retry = true;
+        BookieServerAndResources bookieServerAndResources = null;
+        while (count < retries && retry) {
+            try {
+                log.info("Starting bookie with retry value of :" + count);
+                bookieServerAndResources = runBookie(bkPort);
+                retry = false;
+            } catch (BindException e) {
+                if ( ++count >= retries ) {
+                    e.printStackTrace();
```

Comment:
where does this print? console?

Reply:
Yes, this prints to stderr which will not go to the log, and hence will just be lost. Don't do this.

Reply:
will be removing it, added it for testing purpose and is not required.